### DISCUSSION TITLE
Din rail solution for tasmota devices

### DISCUSSION THREAD

**Discussion by Th0maz** [arendst/Tasmota#18255] [Q&A] [answered]
Hey everybody,
what are your solutions to place an ESP8266 or ESP32 device on a din rail? What kind of housing do you use?
I'd like to connect to the s0 interface of an energy meter in my electricity meter cabinet.
There's a power socket available to use e.g. a 5V USB power supply.

**Comment by barbudor** [comment, marked as answer]
If you search AliExpress for 'din rail enclosure' you will get a lot of proposal for different size and options
If you have a 3D printer, Thingiverse has quite a few models.

If you want a ready device, the ASTRA Rocket Controller can do the job and would provide you with 4 relay and 4 inputs in addition to it's serial port and i2c port

**Reply by VerreyckenGerd** [reply]
I DIY'ed my own: 
![20220606_200558](https://user-images.githubusercontent.com/69856911/227934663-00b6043e-8e42-4b47-9393-29b855814fc8.jpg)

**Comment by HofeBY** [comment]
Rocketcontroller.com with Tasmota

**Comment by Th0maz** [comment]
Thanks for your impressions and ideas.
I think I'll give the rocket controller a try.

**Comment by murfett-au** [comment]
https://www.aliexpress.com/item/1005009328822946.html

**Reply by Noschvie** [reply]
Does this device have an ESP32 chip?
